tests/test_save_load.py

```diff
@@ -375,6 +375,7 @@ def test_warn_buffer(recwarn, model_class, optimize_memory_usage):
         select_env(model_class),
         buffer_size=100,
         optimize_memory_usage=optimize_memory_usage,
+        replay_buffer_kwargs={"handle_timeout_termination": False},
```

Comment:
Hmm why this change? If removed, we might not be testing `handle_timeout_termination` anymore. Ideally tests should use the default values.

Reply:
This change is required because `test_warn_buffer()` runs with `optimize_memory_usage=False` and `optimize_memory_usage=True` (it is designed to test different behaviours depending on the different values of that parameter). In the latter case, the default value of `handle_timeout_termination=True` will cause the `ReplayBuffer` to throw an exception (which is the new behaviour this PR implements). 

I agree it's not ideal we have to change the default values to be able to run this test. However, I think that is inevitable unless we properly fix the bug in #934, or change the default value of `handle_timeout_termination` to `False` (which I think is also not ideal).